streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -390,71 +384,28 @@ private void handleCloseAndRecycle(final Set<Task> tasksToRecycle,
         }
 
         tasksToRecycle.removeAll(tasksToCloseDirty);
-        for (final Task task : tasksToRecycle) {
+        for (final Task oldTask : tasksToRecycle) {
             final Task newTask;
             try {
-                if (task.isActive()) {
-                    final Set<TopicPartition> partitions = standbyTasksToCreate.remove(task.id());
-                    newTask = standbyTaskCreator.createStandbyTaskFromActive((StreamTask) task, partitions);
-                    cleanUpTaskProducer(task, taskCloseExceptions);
+                if (oldTask.isActive()) {
+                    final Set<TopicPartition> partitions = standbyTasksToCreate.remove(oldTask.id());
+                    tasks.convertActiveToStandby((StreamTask) oldTask, partitions, taskCloseExceptions);
```

Comment:
Renamed old `createStandbyTaskFromActive` to `convertActiveToStandby` (same below for reverse conversion)

Reply:
Much better name :)